bpf/custom/bpf_custom.c

```diff
@@ -0,0 +1,29 @@
+// SPDX-License-Identifier: (GPL-2.0-only OR BSD-2-Clause)
+/* Copyright (C) 2020 Authors of Cilium */
+
+#include <bpf/ctx/skb.h>
+#include <bpf/api.h>
+
+#include "lib/common.h"
+
+/* Include custom header file (.h) containing the implementation for
+ * custom_prog(), for example:
+ *
+ *     #include "bytecount.h"
+ */
+#include "bytecount.h"
+
+__section("custom")
+int custom_hook(struct __ctx_buff *ctx)
+{
+	int ret = ctx_load_meta(ctx, CB_CUSTOM_CALLS);
+	int identity = get_identity(ctx);
```

Comment:
This retrieves the identity from the packet mark. AFAIU, we're not guaranteed to have encoded the identity there. I'm unsure what the impact is here.

If this is specific to the current byte counter use case, maybe we should move this into `custom_prog()`? If we want to always provide the security identity to the custom program (i.e., users can assume it's part of the API), maybe we could pass the identity via metadata instead?

Reply:
I believe it will be helpful to collect metrics and I would like to have it part of the API somehow. I did not realise the identity might not always be encoded, thanks. I'll explore ways to pass it through the `cb` buffer, although one concern is to find an offset in `cb` which is unused at this stage. I'm using one for passing the datapath return value already (although we only use a few possible return values I think, so maybe I could encode several values on the 32bits of one `cb` cell if I don't come up with a better solution).